docs/platforms/javascript/guides/nestjs/profiling/index.mdx

```diff
@@ -0,0 +1,92 @@
+---
+
+By default, Sentry error events will not get trace context unless you configure the scope with the transaction, as illustrated in the example below.
+
+<Note>
+
+If you're adopting Profiling in a high-throughput environment, we recommend testing prior to deployment to ensure that your service's performance characteristics maintain expectations.
+
+</Note>
+
+## Installation
+
+<Note>
+
+Node profiling is available starting in `@sentry/profiling-node` version `0.3.0`.
+You have to have the `@sentry/node` (minimum version `7.44.1`) package installed.
+
+</Note>
+
+```bash {tabTitle:npm}
+npm install --save @sentry/node @sentry/profiling-node
+```
+
+```bash {tabTitle:Yarn}
+yarn add @sentry/node @sentry/profiling-node
+```
+
+## Enabling Profiling
+
+To enable profiling, import @sentry/profiling-node, add ProfilingIntegration to your integrations, and set the profilesSampleRate.
+
+<SignInNote />
+
+```javascript
+const Sentry = require("@sentry/node");
+const { nodeProfilingIntegration } = require("@sentry/profiling-node");
+
+Sentry.init({
+  dsn: "___PUBLIC_DSN___",
+  integrations: [
+    // Add our Profiling integration
+    nodeProfilingIntegration(),
+  ],
+  tracesSampleRate: 1.0,
+  // Set sampling rate for profiling - this is relative to tracesSampleRate
+  profilesSampleRate: 1.0,
+});
+
+// Profiling happens automatically after setting it up with `Sentry.init()`.
+// All spans captured in your application will have profiling data attached to them.
+// You can also manually capture spans with `startSpan`, as shown below:
+Sentry.startSpan(
+  {
+    op: "rootSpan",
+    name: "My root span",
+  },
+  () => {
+    // Any code in this callback will be profiled.
+  }
+);
+```
+
+## How Does It Work?
+
+Under the hood, the Sentry profiler uses V8's [CpuProfiler](https://v8docs.nodesource.com/node-18.2/d2/d34/classv8_1_1_cpu_profiler.html) to collect stack samples. This means that sentry/profiling-node is written as a [native add-on](https://nodejs.org/docs/latest-v18.x/api/addons.html) for Node and won't run in environments like Deno or Bun. Profiling enhances tracing by providing profiles for individual transactions. This allows you to look at higher level performance information like transaction and span durations before diving deeper and looking at profiles.
+
+## Runtime Flags
+
+The default mode of the v8 CpuProfiler is [kEagerLogging](https://v8docs.nodesource.com/node-18.2/d2/dc3/namespacev8.html#a874b4921ddee43bef58d8538e3149374) which enables the profiler even when no profiles are active - this is good because it makes calls to startProfiling fast at the tradeoff for constant CPU overhead. The behavior can be controlled via the SENTRY_PROFILER_LOGGING_MODE environment variable with values of eager|lazy. If you opt to use the lazy logging mode, calls to startProfiling may be slow (depending on environment and node version, it can be in the order of a few hundred ms).
```

Comment:
```suggestion
The default mode of the v8 CpuProfiler is [kEagerLogging](https://v8docs.nodesource.com/node-18.2/d2/dc3/namespacev8.html#a874b4921ddee43bef58d8538e3149374), which enables the profiler even when no profiles are active — this is good because it makes calls to `startProfiling` fast with the tradeoff of constant CPU overhead. The behavior can be controlled via the `SENTRY_PROFILER_LOGGING_MODE` environment variable with values of `eager|lazy`. If you opt to use the lazy logging mode, calls to `startProfiling` may be slow. (Depending on environment and node version, it can be in the order of a few hundred ms.)
```